### PR TITLE
Move pluginFactory out of ProxyConnection

### PR DESCRIPTION
**Pull request by vogdb**
Right after #52 is completed

### CODE REVIEW

**Comment by vogdb** [comment]
@fvovan please review

File: lib/plugin/plugin-registry.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  pluginClassList.forEach(function(plugin) {
+    this.registerPlugin(plugin);
+  }.bind(this));
+}
+
+/**
+ * @param {Function} pluginClass
+ */
+PluginRegistry.prototype.registerPlugin = function(pluginClass) {
+  if (!(pluginClass instanceof Function)) {
```
Comment (fvovan) [inline lib/plugin/plugin-registry.js]: Shouldn't it be an instance of `PluginAbstract` ?

File: lib/plugin/plugin-registry.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+PluginRegistry.prototype.isAllowedPlugin = function(type) {
+  return !!this._types[type];
+};
+
+PluginRegistry.prototype.createPlugin = function(type, id) {
+  var pluginClass = this._types[type];
+  if (!pluginClass) {
+    throw new Error('Invalid plugin');
+  }
+  return new pluginClass(id, type, this);
```
Comment (fvovan) [inline lib/plugin/plugin-registry.js]: Third argument of  `PluginAbstract` is `proxyConnection` but here you pass `PluginRegistry ` as `this`
> `function PluginAbstract(id, type, proxyConnection)`

File: lib/plugin/plugin-registry.js (hunk 1 above)
Comment (vogdb) [inline lib/plugin/plugin-registry.js]: It's long to explain. The code shows it better
```js
var util = require('util');

function A() {
}
function B() {
  A.apply(this, arguments)
}
util.inherits(B, A);

console.log(B instanceof A);// false
var b = new B();
console.log(b instanceof A);// true
```

In `registerPlugin` we are waiting for class not instance of class

Comment (fvovan) [inline lib/plugin/plugin-registry.js]: Thanks. I realised it. Actually I updated my comment but probably it was too late.

**Comment by fvovan** [comment]
the rest looks okay

**Comment by vogdb** [comment]
@fvovan please review

**Comment by vogdb** [comment]
@fvovan please review

**Comment by fvovan** [comment]
lgtm

**Pull request merged**